src/v0/destinations/tiktok_ads/util.js

```diff
@@ -60,4 +60,29 @@ const hashUserField = (user) => {
   }
   return updatedUser;
 };
-module.exports = { getContents, hashUserField };
+
+/**
+ * Get event source from message properties or channel. We have considered the following doc
+ * https://business-api.tiktok.com/portal/docs?id=1771100779668482
+ * @param {*} message
+ * @returns (string): The resolved, valid event source — one of 'web', 'app', 'offline', or 'crm'. Defaults to 'web' if none are valid.
+ */
+const getEventSource = ({ properties, channel }) => {
+  const supportedSources = ['web', 'app', 'offline', 'crm'];
+
+  const isSupported = (source) => supportedSources.includes(source);
+
+  const { eventSource } = properties || {};
```

Comment:
we need to refer here rudderstack docs also. I could find where are we documenting that in properties we need to put eventSource?

Reply:
Yes, we have to update our doc, as I mentioned in the pr description. I have already checked with the product regarding this mapping. You can check the Slack thread for more information.